File: components/api/api-modules-javascript/src/main/resources/META-INF/dirigible/modules/src/utils/qrcode.ts
/*
 * Copyright (c) 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */

import * as bytes from "sdk/io/bytes";
const QRCodeFacade = Java.type("org.eclipse.dirigible.components.api.utils.QRCodeFacade");

export class QRCode {

    public static generateQRCode(text: string): any[] {
        return bytes.toJavaScriptBytes(QRCodeFacade.generateQRCode(text));
    }
}

// @ts-ignore
if (typeof module !== 'undefined') {
    // @ts-ignore
    module.exports = QRCode;
}

File: components/api/api-modules-java/src/main/java/org/eclipse/dirigible/sdk/utils/QrCode.java
/*
 * Copyright (c) 2010-2026 Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: Eclipse Dirigible contributors SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.dirigible.sdk.utils;

import com.google.zxing.WriterException;
import java.io.IOException;
import org.eclipse.dirigible.components.api.utils.QRCodeFacade;

/**
 * Renders a QR-code PNG for the given payload. Backed by ZXing under the hood with the platform's
 * default size and error-correction settings; ideal for embedding small URLs, codes, or connection
 * strings in receipts, emails, and IDE perspectives.
 * <p>
 * Returns the raw PNG bytes — write them straight to an HTTP response (with
 * {@code Content-Type: image/png}) or into a file via {@code Files.writeBytesNative}.
 */
public final class QrCode {

    private QrCode() {}

    public static byte[] generate(String text) throws WriterException, IOException {
        return QRCodeFacade.generateQRCode(text);
    }
}
